### PR TITLE
Do hash checks for bundle add

### PR DESCRIPTION
**Pull request by tmarcu**
Hash check every file before adding it to the file system.

Signed-off-by: Tudor Marcu <tudor.marcu@intel.com>

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -721,6 +721,22 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
...
+	iter = list_head(to_install_files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+
+		string_or_die(&fullname, "%s/staged/%s", state_dir, file->hash);
+
+		ret = verify_file(file, fullname);
+		if (!ret) {
+			free(fullname);
```
Comment (phmccarty) [inline src/bundle.c]: you'll want to free() after the fprintf() below, not before it

File: src/bundle.c [hunk 2]
```diff
@@ -721,6 +721,22 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
...
+			fprintf(stderr, "FAILED HASH CHECK %s\n", fullname);
```
Comment (phmccarty) [inline src/bundle.c]: I would prefer this message to be more consistent with other errors (i.e. "Error: hash check failed for %s\n")

**Comment by matthewrsj** [comment]
FYI @tmarcu and @phmccarty I'm working on fixing the tests. It looks like a few code changes need to be made. I can make those if you don't mind @tmarcu .

**Comment by phmccarty** [comment]
+1

**Comment by tmarcu** [comment]
+1 from the past

**Pull request merged**